src/cmakeTaskProvider.ts

```diff
@@ -224,31 +252,51 @@ class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.OutputConsu
     }
 
     private async runConfigTask(): Promise<any> {
-        this.writeEmitter.fire(localize("config.started", "Config Started...") + endOfLine);
-        const result: number | undefined =  await vscode.commands.executeCommand('cmake.configure');
-        this.closeEmitter.fire(result ? result : -1);
+        this.writeEmitter.fire(localize("config.started", "Config task started...") + endOfLine);
+        let result: number | undefined | null;
+        if (this.preset) {
+            const cmakeTools: CMakeTools | undefined = getCMakeToolsForActiveFolder();
+            const cmakeDriver: CMakeDriver | undefined = (await cmakeTools?.getCMakeDriverInstance()) || undefined;
+            const cmakePath: string = cmakeDriver?.getCMakeCommand() || "CMake.EXE";
+            const args: string[] = [ "--preset", this.preset];
+            const execResult = await proc.execute(cmakePath, args, this, this.options).result;
+            result = execResult?.retc;
+        } else {
+            result =  await vscode.commands.executeCommand('cmake.configure');
```

Comment:
Since we are adding options to set the environment and cwd for the task, we need to figure out a way to make it apply to all the commands.  It will be very confusing to people if they set it and it only works for presets.

And now that I'm thinking about it more, I'm wondering if we should be treating presets differently here vs just calling `executeCommand` and letting the extension do the rest. Part of the value of the extension is that it can support presets even for versions of cmake that don't.